lib/api/BackbeatAPI.js

```diff
@@ -1171,7 +1261,18 @@ class BackbeatAPI {
             schedule.setHours(schedule.getHours() + hours);
         }
 
-        sites.forEach(site => {
+        return async.eachLimit(sites, 10, (site, next) => {
+            if (service === 'lifecycle') {
```

Comment:
we have three services here and doing different kinds of things for each, just my personal thinking, maybe a `switch(service) case` logic is more clear?